app/test/vesting_test.go

```diff
@@ -0,0 +1,550 @@
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+	tmtime "github.com/tendermint/tendermint/types/time"
+	"google.golang.org/grpc"
+)
+
+func TestVestingModule(t *testing.T) {
+	suite.Run(t, new(VestingModuleTestSuite))
+}
+
+type VestingModuleTestSuite struct {
+	suite.Suite
+
+	// Regular accounts
+	accounts       []string
+	mut            sync.Mutex
+	accountCounter int
+}
+
+func (s *VestingModuleTestSuite) SetupSuite() {
+	t := s.T()
+	t.Log("setting up vesting module test suite")
+	s.accounts = testfactory.GenerateAccounts(300)
+}
+
+func (s *VestingModuleTestSuite) unusedAccount() string {
+	s.mut.Lock()
+	acc := s.accounts[s.accountCounter]
+	s.accountCounter++
+	s.mut.Unlock()
+	return acc
+}
+
+func (s *VestingModuleTestSuite) NewNetworkWithGenesisOpt(kr keyring.Keyring, genesisOpts ...testnode.GenesisOption) (cctx testnode.Context, rpcAddr, grpcAddr string) {
+	tmCfg := testnode.DefaultTendermintConfig()
+	tmCfg.Consensus.TargetHeightDuration = time.Millisecond * 1
+	appConf := testnode.DefaultAppConfig()
+
+	for _, name := range s.accounts {
+		testfactory.NewBaseAccount(kr, name)
+	}
+
+	cctx, rpcAddr, grpcAddr = testnode.NewNetwork(s.T(), testnode.DefaultParams(), tmCfg, appConf, []string{}, genesisOpts...)
+	cctx.Keyring = kr
+
+	return cctx, rpcAddr, grpcAddr
+}
+
+func (s *VestingModuleTestSuite) TestGenesisDelayedVestingAccounts() {
+	// initial unlocked allocation to pay the gas fees
+	initBalanceForGasFee := sdk.NewCoin(app.BondDenom, sdk.NewInt(10))
+
+	tests := []struct {
+		name    string
+		accName string // will be filled by the code
+		endTime time.Time
+	}{
+		{
+			name:    "ValidVestingAccount",
+			endTime: time.Now().Add(time.Hour * 50),
+		},
+		{
+			name:    "ImmediateVesting",
+			endTime: time.Now(),
+		},
+		{
+			name:    "NegativeEndTime",
+			endTime: time.Now().Add(-time.Second * 10),
+		},
+	}
+
+	kr := testfactory.GenerateKeyring()
+
+	gsOpt := func(gs map[string]json.RawMessage) map[string]json.RawMessage {
+		vAccs := authtypes.GenesisAccounts{}
+		vBals := []banktypes.Balance{}
+		for i, tt := range tests {
+			tests[i].accName = fmt.Sprintf("vesting_%d", i)
+			bacc, coins := testfactory.NewBaseAccount(kr, tests[i].accName)
+			acc := vestingtypes.NewDelayedVestingAccount(bacc, coins, tt.endTime.Unix())
+			vAccs = append(vAccs, acc)
+			vBals = append(vBals, banktypes.Balance{
+				Address: acc.GetAddress().String(),
+				Coins:   acc.GetOriginalVesting().Add(initBalanceForGasFee),
+			})
+		}
+
+		var err error
+		gs, err = AddAccountsToGenesisState(gs, vAccs...)
+		assert.NoError(s.T(), err)
+
+		gs, err = AddBalancesToGenesisState(gs, vBals...)
+		assert.NoError(s.T(), err)
+
+		return gs
+	}
+
+	cctx, _, _ := s.NewNetworkWithGenesisOpt(kr, gsOpt)
+
+	assert.NoError(s.T(), cctx.WaitForNextBlock())
+
+	for _, tt := range tests {
+		s.Run(tt.name, func() {
+			accAddress := getAddress(tt.accName, cctx.Keyring).String()
+
+			// Test account details correctness
```

Comment:
it tests if the given genesis data is read and processed correctly into the desired type of account.
They are implicitly tested in the lines below that, so I am ok with removing those.
@cmwaters and @evan-forbes do you see any benefit in keeping them?

Reply:
Clarification: the suggestion is to remove the comment on line 126. I think lines 128-132 are fine but since they don't verify any details inside an account, I don't think the comment is necessary.